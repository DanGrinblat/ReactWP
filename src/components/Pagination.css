.pagination {
	margin-bottom: 74px !important;
}

