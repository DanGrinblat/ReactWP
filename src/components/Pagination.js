import "./Pagination.css";
import React from "react";

class Pagination extends React.Component {
	onClickPageHelper = event => {
		console.log(event);
		this.props.onClickPage(event.target.id);
	};

	renderPageButtons() {
		const pageNumbers = [];
		var pageButtons = [];

		for (var i = 1; i <= this.props.pageCount; i++) {
			pageNumbers.push(i);
		}

		pageButtons = pageNumbers.map(number => {
			return (
				<a
					key={number}
					id={number}
					className={`item ${
						this.props.selectedPage === number ? "active" : ""
					}`}
					onClick={this.onClickPageHelper}
				>
					{number}
				</a>
			);
		});

		return pageButtons;
	}

	render() {
		return (
			<div className="ui pagination menu">{this.renderPageButtons()}</div>
		);
	}
}

export default Pagination;
