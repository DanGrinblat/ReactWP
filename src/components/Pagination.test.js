import React from "react";
import { shallow } from "enzyme";
import Pagination from "./Pagination";

describe("Pagination", () => {
	const mockOnClickPage = jest.fn();
	const props = {
		pageCount: 2,
		selectedPage: 1,
		onClickPage: mockOnClickPage
	};
	const pagination = shallow(<Pagination {...props} />);

	it("should render correctly", () => {
		expect(pagination).toMatchSnapshot();
	});

	describe("when clicking a page", () => {
		it("should call onClickPage in the props", () => {
			pagination
				.find(".item")
				.at(1)
				.simulate("click", { target: { value: 1 } });
			expect(mockOnClickPage).toHaveBeenCalled();
		});
	});
});
