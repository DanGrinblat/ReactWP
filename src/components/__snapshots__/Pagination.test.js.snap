// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination should render correctly 1`] = `
<div
  className="ui pagination menu"
>
  <a
    className="item active"
    id={1}
    key="1"
    onClick={[Function]}
  >
    1
  </a>
  <a
    className="item "
    id={2}
    key="2"
    onClick={[Function]}
  >
    2
  </a>
</div>
`;
